compiler/src/dotty/tools/dotc/transform/ExplicitOuter.scala

```diff
@@ -49,7 +50,7 @@ class ExplicitOuter extends MiniPhase with InfoTransformer { thisPhase =>
   /** Add outer accessors if a class always needs an outer pointer */
   override def transformInfo(tp: Type, sym: Symbol)(using Context): Type = tp match {
     case tp @ ClassInfo(_, cls, _, decls, _) if needsOuterAlways(cls) =>
-      val newDecls = decls.cloneScope
+      val newDecls = if decls == EmptyScope then newScope else decls.cloneScope
```

Comment:
There's multiple points in the compiler where we do decls.cloneScope. I don't really see why we couldn't call cloneScope on an EmptyScope, so maybe we should just change the implementation of EmptyScope.cloneScope to return `newScope(EmptyScope.nestingLevel)` instead of crashing.